### PR TITLE
Forms refactoring - Tidy up

### PR DESCRIPTION
**Pull request by cintamani**
From: https://eaflood.atlassian.net/browse/RUBY-693

There are some minor tidy-ups in this PR:
 - Get rid of `initialize` implementation on children forms that only call `super`
 - Make sure we always access `transient_registration` through its `attr_accessor` method
 - Move validations declaration to the top of the file, between class methods definitions and instance methods definitions

### CODE REVIEW

File: app/forms/waste_carriers_engine/contact_email_form.rb [hunk 1]
```diff
@@ -4,10 +4,13 @@ module WasteCarriersEngine
...
     attr_accessor :contact_email, :confirmed_email
 
+    validates :contact_email, :confirmed_email, "defra_ruby/validators/email": true
+    validates :confirmed_email, "waste_carriers_engine/matching_email": { compare_to: :contact_email }
+
     def initialize(transient_registration)
       super
-      self.contact_email = @transient_registration.contact_email
-      self.confirmed_email = @transient_registration.contact_email
+      self.contact_email = transient_registration.contact_email
```
Comment (Cruikshanks) [inline app/forms/waste_carriers_engine/contact_email_form.rb]: NIT Adding a line between super and this as per other changes

```suggestion
      
      self.contact_email = transient_registration.contact_email
```

File: app/forms/waste_carriers_engine/contact_name_form.rb [hunk 2]
```diff
@@ -4,10 +4,12 @@ module WasteCarriersEngine
   class ContactNameForm < BaseForm
     attr_accessor :first_name, :last_name
 
+    validates :first_name, :last_name, "waste_carriers_engine/person_name": true
+
     def initialize(transient_registration)
       super
-      self.first_name = @transient_registration.first_name
-      self.last_name = @transient_registration.last_name
+      self.first_name = transient_registration.first_name
```
Comment (Cruikshanks) [inline app/forms/waste_carriers_engine/contact_name_form.rb]: NIT Adding a line between super and this as per other changes

```suggestion
      
      self.first_name = transient_registration.first_name
```

File: app/forms/waste_carriers_engine/contact_phone_form.rb [hunk 3]
```diff
@@ -4,9 +4,11 @@ module WasteCarriersEngine
   class ContactPhoneForm < BaseForm
     attr_accessor :phone_number
 
+    validates :phone_number, "defra_ruby/validators/phone_number": true
+
     def initialize(transient_registration)
       super
-      self.phone_number = @transient_registration.phone_number
+      self.phone_number = transient_registration.phone_number
```
Comment (Cruikshanks) [inline app/forms/waste_carriers_engine/contact_phone_form.rb]: NIT Adding a line between super and this as per other changes

```suggestion
      
      self.phone_number = transient_registration.phone_number
```

File: app/forms/waste_carriers_engine/contact_postcode_form.rb [hunk 4]
```diff
@@ -4,9 +4,11 @@ module WasteCarriersEngine
   class ContactPostcodeForm < PostcodeForm
     attr_accessor :temp_contact_postcode
 
+    validates :temp_contact_postcode, "waste_carriers_engine/postcode": true
+
     def initialize(transient_registration)
       super
-      self.temp_contact_postcode = @transient_registration.temp_contact_postcode
+      self.temp_contact_postcode = transient_registration.temp_contact_postcode
```
Comment (Cruikshanks) [inline app/forms/waste_carriers_engine/contact_postcode_form.rb]: NIT Adding a line between super and this as per other changes

```suggestion
      
      self.temp_contact_postcode = transient_registration.temp_contact_postcode
```

File: app/forms/waste_carriers_engine/declaration_form.rb [hunk 5]
```diff
@@ -8,9 +8,11 @@ def self.can_navigate_flexibly?
       false
     end
 
+    validates :declaration, inclusion: { in: [1] }
+
     def initialize(transient_registration)
       super
-      self.declaration = @transient_registration.declaration
+      self.declaration = transient_registration.declaration
```
Comment (Cruikshanks) [inline app/forms/waste_carriers_engine/declaration_form.rb]: NIT Adding a line between super and this as per other changes

```suggestion
      
      self.declaration = transient_registration.declaration
```

File: app/forms/waste_carriers_engine/declare_convictions_form.rb [hunk 6]
```diff
@@ -4,9 +4,11 @@ module WasteCarriersEngine
   class DeclareConvictionsForm < BaseForm
     attr_accessor :declared_convictions
 
+    validates :declared_convictions, "waste_carriers_engine/yes_no": true
+
     def initialize(transient_registration)
       super
-      self.declared_convictions = @transient_registration.declared_convictions
+      self.declared_convictions = transient_registration.declared_convictions
```
Comment (Cruikshanks) [inline app/forms/waste_carriers_engine/declare_convictions_form.rb]: NIT Adding a line between super and this as per other changes

```suggestion
      
      self.declared_convictions = transient_registration.declared_convictions
```

File: app/forms/waste_carriers_engine/location_form.rb [hunk 7]
```diff
@@ -4,9 +4,11 @@ module WasteCarriersEngine
   class LocationForm < BaseForm
     attr_accessor :location
 
+    validates :location, "defra_ruby/validators/location": { allow_overseas: true }
+
     def initialize(transient_registration)
       super
-      self.location = @transient_registration.location
+      self.location = transient_registration.location
```
Comment (Cruikshanks) [inline app/forms/waste_carriers_engine/location_form.rb]: NIT Adding a line between super and this as per other changes

```suggestion
      
      self.location = transient_registration.location
```

File: app/forms/waste_carriers_engine/main_people_form.rb [hunk 8]
```diff
@@ -6,29 +6,29 @@ class MainPeopleForm < PersonForm
 
     attr_accessor :business_type
 
+    validates_with MainPersonValidator
+
     def initialize(transient_registration)
       super
       # We only use this for the correct microcopy
```
Comment (Cruikshanks) [inline app/forms/waste_carriers_engine/main_people_form.rb]: NIT Adding a line between super and this as per other changes

```suggestion
      
      # We only use this for the correct microcopy
```

File: app/forms/waste_carriers_engine/manual_address_form.rb [hunk 9]
```diff
@@ -8,15 +8,23 @@ class ManualAddressForm < BaseForm
...
+    validates :address_line_1, presence: true, length: { maximum: 160 }
+    validates :address_line_2, length: { maximum: 70 }
+    validates :town_city, presence: true, length: { maximum: 30 }
+    validates :postcode, length: { maximum: 30 }
+    validates :country, presence: true, if: :overseas?
+    validates :country, length: { maximum: 50 }
+
     def initialize(transient_registration)
       super
       # We use this for the correct microcopy and to determine what fields to show
```
Comment (Cruikshanks) [inline app/forms/waste_carriers_engine/manual_address_form.rb]: NIT Adding a line between super and this as per other changes

```suggestion
      
      # We use this for the correct microcopy and to determine what fields to show
```

File: app/forms/waste_carriers_engine/other_businesses_form.rb [hunk 10]
```diff
@@ -4,9 +4,11 @@ module WasteCarriersEngine
   class OtherBusinessesForm < BaseForm
     attr_accessor :other_businesses
 
+    validates :other_businesses, "waste_carriers_engine/yes_no": true
+
     def initialize(transient_registration)
       super
-      self.other_businesses = @transient_registration.other_businesses
+      self.other_businesses = transient_registration.other_businesses
```
Comment (Cruikshanks) [inline app/forms/waste_carriers_engine/other_businesses_form.rb]: NIT Adding a line between super and this as per other changes

```suggestion
      
      self.other_businesses = transient_registration.other_businesses
```

File: app/forms/waste_carriers_engine/payment_summary_form.rb [hunk 11]
```diff
@@ -4,18 +4,20 @@ module WasteCarriersEngine
...
+    validates :temp_payment_method, inclusion: { in: %w[card bank_transfer] }
+
     def self.can_navigate_flexibly?
       false
     end
 
     def initialize(transient_registration)
       super
-      self.temp_payment_method = @transient_registration.temp_payment_method
+      self.temp_payment_method = transient_registration.temp_payment_method
```
Comment (Cruikshanks) [inline app/forms/waste_carriers_engine/payment_summary_form.rb]: NIT Adding a line between super and this as per other changes

```suggestion
      
      self.temp_payment_method = transient_registration.temp_payment_method
```

File: app/forms/waste_carriers_engine/registration_number_form.rb [hunk 12]
```diff
@@ -4,11 +4,13 @@ module WasteCarriersEngine
   class RegistrationNumberForm < BaseForm
     attr_accessor :company_no, :business_type
 
+    validates :company_no, "defra_ruby/validators/companies_house_number": true
+
     def initialize(transient_registration)
       super
-      self.company_no = @transient_registration.company_no
+      self.company_no = transient_registration.company_no
```
Comment (Cruikshanks) [inline app/forms/waste_carriers_engine/registration_number_form.rb]: NIT Adding a line between super and this as per other changes

```suggestion
      
      self.company_no = transient_registration.company_no
```

File: app/forms/waste_carriers_engine/renewal_complete_form.rb [hunk 13]
```diff
@@ -11,9 +11,9 @@ def self.can_navigate_flexibly?
     def initialize(transient_registration)
       super
       self.certificate_link = build_certificate_link
```
Comment (Cruikshanks) [inline app/forms/waste_carriers_engine/renewal_complete_form.rb]: NIT Adding a line between super and this as per other changes

```suggestion
      
      self.certificate_link = build_certificate_link
```

File: app/forms/waste_carriers_engine/renewal_information_form.rb [hunk 14]
```diff
@@ -6,8 +6,8 @@ class RenewalInformationForm < BaseForm
 
     def initialize(transient_registration)
       super
-      self.type_change = @transient_registration.registration_type_changed?
-      self.total_fee = @transient_registration.fee_including_possible_type_change
+      self.type_change = transient_registration.registration_type_changed?
```
Comment (Cruikshanks) [inline app/forms/waste_carriers_engine/renewal_information_form.rb]: NIT Adding a line between super and this as per other changes

```suggestion
      
      self.type_change = transient_registration.registration_type_changed?
```

File: app/forms/waste_carriers_engine/renewal_received_form.rb [hunk 15]
```diff
@@ -10,10 +10,10 @@ def self.can_navigate_flexibly?
 
     def initialize(transient_registration)
       super
-      self.contact_email = @transient_registration.contact_email
-      self.pending_convictions_check = @transient_registration.conviction_check_required?
-      self.pending_payment = @transient_registration.pending_payment?
-      self.pending_worldpay_payment = @transient_registration.pending_worldpay_payment?
+      self.contact_email = transient_registration.contact_email
```
Comment (Cruikshanks) [inline app/forms/waste_carriers_engine/renewal_received_form.rb]: NIT Adding a line between super and this as per other changes

```suggestion
      
      self.contact_email = transient_registration.contact_email
```

File: app/forms/waste_carriers_engine/service_provided_form.rb [hunk 16]
```diff
@@ -4,9 +4,11 @@ module WasteCarriersEngine
   class ServiceProvidedForm < BaseForm
     attr_accessor :is_main_service
 
+    validates :is_main_service, "waste_carriers_engine/yes_no": true
+
     def initialize(transient_registration)
       super
-      self.is_main_service = @transient_registration.is_main_service
+      self.is_main_service = transient_registration.is_main_service
```
Comment (Cruikshanks) [inline app/forms/waste_carriers_engine/service_provided_form.rb]: NIT Adding a line between super and this as per other changes

```suggestion
      
      self.is_main_service = transient_registration.is_main_service
```

File: app/forms/waste_carriers_engine/tier_check_form.rb [hunk 17]
```diff
@@ -4,9 +4,11 @@ module WasteCarriersEngine
   class TierCheckForm < BaseForm
     attr_accessor :temp_tier_check
 
+    validates :temp_tier_check, "waste_carriers_engine/yes_no": true
+
     def initialize(transient_registration)
       super
-      self.temp_tier_check = @transient_registration.temp_tier_check
+      self.temp_tier_check = transient_registration.temp_tier_check
```
Comment (Cruikshanks) [inline app/forms/waste_carriers_engine/tier_check_form.rb]: NIT Adding a line between super and this as per other changes

```suggestion
      
      self.temp_tier_check = transient_registration.temp_tier_check
```

File: app/forms/waste_carriers_engine/waste_types_form.rb [hunk 18]
```diff
@@ -4,9 +4,11 @@ module WasteCarriersEngine
   class WasteTypesForm < BaseForm
     attr_accessor :only_amf
 
+    validates :only_amf, "waste_carriers_engine/yes_no": true
+
     def initialize(transient_registration)
       super
-      self.only_amf = @transient_registration.only_amf
+      self.only_amf = transient_registration.only_amf
```
Comment (Cruikshanks) [inline app/forms/waste_carriers_engine/waste_types_form.rb]: NIT Adding a line between super and this as per other changes

```suggestion
      
      self.only_amf = transient_registration.only_amf
```

File: app/forms/waste_carriers_engine/business_type_form.rb [hunk 19]
```diff
@@ -4,9 +4,11 @@ module WasteCarriersEngine
   class BusinessTypeForm < BaseForm
     attr_accessor :business_type
 
+    validates :business_type, "defra_ruby/validators/business_type": { allow_overseas: true }
+
     def initialize(transient_registration)
       super
-      self.business_type = @transient_registration.business_type
+      self.business_type = transient_registration.business_type
```
Comment (cintamani) [inline app/forms/waste_carriers_engine/business_type_form.rb]:
```suggestion

      self.business_type = transient_registration.business_type
```

File: app/forms/waste_carriers_engine/main_people_form.rb (hunk 8 above)
Comment (cintamani) [inline app/forms/waste_carriers_engine/main_people_form.rb]:
```suggestion

      # We only use this for the correct microcopy
```

File: app/forms/waste_carriers_engine/person_form.rb [hunk 20]
```diff
@@ -5,6 +5,8 @@ class PersonForm < BaseForm
     attr_accessor :first_name, :last_name, :position, :dob_day, :dob_month, :dob_year, :dob
     attr_accessor :new_person
 
+    validate :old_enough?
+
     def initialize(transient_registration)
```
Comment (cintamani) [inline app/forms/waste_carriers_engine/person_form.rb]:
```suggestion
```

File: app/forms/waste_carriers_engine/person_form.rb [hunk 21]
```diff
@@ -5,6 +5,8 @@ class PersonForm < BaseForm
...
       super
```
Comment (cintamani) [inline app/forms/waste_carriers_engine/person_form.rb]:
```suggestion
```

File: app/forms/waste_carriers_engine/person_form.rb [hunk 22]
```diff
@@ -5,6 +5,8 @@ class PersonForm < BaseForm
...
     end
```
Comment (cintamani) [inline app/forms/waste_carriers_engine/person_form.rb]:
```suggestion
```

**Pull request merged**